src/main/java/edu/umich/ctools/sectionsUtilityTool/SISPollingThread.java

```diff
@@ -0,0 +1,221 @@
+		String errmsg = "Problem in sending the email for course %s due to %s";
+		try {
+			message.addRecipient(Message.RecipientType.CC, new InternetAddress(ccAddress));
+			message.addRecipient(Message.RecipientType.TO, new InternetAddress(data.getEmailAddress()));
+			message.setFrom(new InternetAddress(fromAddress));
+			message.setSubject(String.format(data.getSisProcessType().getDescription() + " " + data.getCourseId()));
+
+			//if canvas taking long time to process the sis request, after X attempts intentionally we take the request
+			// out of the thread pool that's when apiResp is null and we send a email to user about this situation.
+			if ((apiResp == null) && data.isSISUploadFailed()) {
+				Multipart multipart = new MimeMultipart();
+				BodyPart bodyParts = new MimeBodyPart();
+				//BSA might come up with better wording, so keeping it simple for now
+				bodyParts.setText((data.getSisProcessType().getDescription() + " " +
+						"request either failed or canvas is taking too long to process this request"));
+				message.setContent(multipart);
+				Transport.send(message);
+				return;
+			}
+
+			JSONObject resp = new JSONObject(apiResp);
+			String workflowState = (String) resp.get(Utils.JSON_PARAM_WORKFLOW_STATE);
+
+			Multipart multipart = new MimeMultipart();
+			String msgBody = getBody(data.getSisProcessType(), resp);
+			//msgBody of the email
+			BodyPart body = new MimeBodyPart();
+			body.setText(msgBody);
+			multipart.addBodyPart(body);
+			//Attachment of sisData
+			if (workflowState.equals(Utils.JSON_PARAM_IMPORTED)) {
+				multipart.addBodyPart(getAttachment(data));
+			}
+
+			message.setContent(multipart);
+			M_log.info("Sending email for course " + data.getCourseId());
+			Transport.send(message);
+		} catch (MessagingException | JSONException e) {
+			M_log.error(String.format(errmsg, data.getCourseId(), e.getMessage()));
+		} catch (Exception e) {
+			M_log.error(String.format(errmsg, data.getCourseId(), e.getMessage()));
+		}
+
+	}
+
+	public static String getBody(SISUploadType sisProcessType, JSONObject apiResp) throws JSONException {
+		String workflowState = (String) apiResp.get(Utils.JSON_PARAM_WORKFLOW_STATE);
+		String startTime = (String) apiResp.get("started_at");
+		String endTime = (String) apiResp.get("ended_at");
+
+		StringBuilder msgBody = new StringBuilder();
+		msgBody.append("StartTime: " + startTime);
+		msgBody.append("\n");
+		msgBody.append("EndTime: " + endTime);
+		msgBody.append("\n");
+
+		if (workflowState.equals(Utils.JSON_PARAM_IMPORTED_WITH_MESSAGES)) {
+			msgBody.append("SIS upload imported with some errors");
+			msgBody.append("\n");
+			JSONArray processing_warnings = apiResp.getJSONArray("processing_warnings");
+			for (int i = 0; i < processing_warnings.length(); i++) {
+				msgBody.append(processing_warnings.get(i));
+				msgBody.append("\n");
+			}
+			return msgBody.toString();
+
+		}
+
+		if (workflowState.equals(Utils.JSON_PARAM_FAILED_WITH_MESSAGES)) {
```

Comment:
Does failed mean "nothing in the upload worked"?  This seems identical to the prior bit of processing except for the top level message.  Better to make this block a method and passing in the top level method.